learning_resources_search/serializers.py

```diff
@@ -10,10 +16,217 @@
+
+    def validate_platform(self, data):
+        if data:
+            for platform_value in data.split(","):
+                if platform_value.lower() not in [
+                    e.value.lower() for e in PlatformType
+                ]:
+                    msg = f"{platform_value} is not a valid option"
+                    raise serializers.ValidationError(msg)
+        return data
+
+    def validate_professional(self, data):
+        if data:
+            for professional_value in data.split(","):
+                if professional_value.lower() not in ["true", "false"]:
+                    msg = f"{professional_value} is not a valid option"
+                    raise serializers.ValidationError(msg)
+        return data
+
+    def validate_aggregations(self, data):
+        if data:
+            for agg_value in data.split(","):
+                if agg_value.lower() not in AGGREGATIONS:
+                    msg = f"{agg_value} is not a valid option"
+                    raise serializers.ValidationError(msg)
+        return data
+
+
+def _transform_aggregations(aggregations):
+    formatted_aggregations = {}
+    for key, value in aggregations.items():
+        if value.get("buckets"):
+            formatted_aggregations[key] = value.get("buckets")
+        elif value.get(key, {}).get("buckets"):
+            formatted_aggregations[key] = value.get(key, {}).get("buckets")
+        else:
+            formatted_aggregations[key] = value.get(key, {}).get(key, {}).get("buckets")
+    return formatted_aggregations
+
+
+def _transform_search_results_suggest(search_result):
+    """
+    Transform suggest results from opensearch
+
+    Args:
+        search_result (dict): The results from opensearch
+
+    Returns:
+        dict: The opensearch response dict with transformed suggestions
+    """
+
+    es_suggest = search_result.pop("suggest", {})
+    if (
+        search_result.get("hits", {}).get("total", {}).get("value")
+        <= settings.OPENSEARCH_MAX_SUGGEST_HITS
+    ):
+        suggestion_dict = defaultdict(int)
+        suggestions = [
+            suggestion
+            for suggestion_list in extract_values(es_suggest, "options")
+            for suggestion in suggestion_list
+            if suggestion["collate_match"] is True
+        ]
+        for suggestion in suggestions:
+            suggestion_dict[suggestion["text"]] = (
+                suggestion_dict[suggestion["text"]] + suggestion["score"]
+            )
+        return [
+            key
+            for key, value in sorted(
+                suggestion_dict.items(), key=lambda item: item[1], reverse=True
+            )
+        ][: settings.OPENSEARCH_MAX_SUGGEST_RESULTS]
+    else:
+        return []
+
+
+class LearningResourcesSearchResponseSerializer(serializers.Serializer):
+    count = serializers.SerializerMethodField()
+    results = serializers.SerializerMethodField()
+    metadata = serializers.SerializerMethodField()
+
+    def get_count(self, instance):
```

Comment:
I believe that `drf-spectacular` is not picking up the response format at all right now.

I think the best way to inform spectacular is to decorate the methods here.
```python
def get_count(self, instance) -> int:
    pass

@extend_schema_field(LearningResourceSerializer(many=True))
def get_results(self, instance):
    pass
```